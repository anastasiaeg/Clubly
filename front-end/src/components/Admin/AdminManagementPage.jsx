import React from "react";
//import './orgmanagement.css';
import {
  Dialog,
  H5,
  InputGroup,
  Button,
  AnchorButton,
  Classes,
  Intent
} from "@blueprintjs/core";

export default class AdminManagementPage extends React.Component {
  emptyOrg = {
    club_name: "",
    description: "",
    tags: [],
    members: [],
    events: [],
    organizers: []
  };

  constructor(props) {
    super(props);
    this.state = {
      orgDialogOpened: false,
      org: this.emptyOrg
    };
  }

  toggleDialog = (club = Object.assign({}, this.emptyOrg)) => {
    this.setState({
      orgDialogOpened: !this.state.orgDialogOpened,
      org: club
    });
  };

  handleCloseDialog = () => {
    this.setState({
      orgDialogOpened: false
    });
  };

  render() {
    const { orgDialogOpened, org } = this.state;
    const wicsData = {
      club_name: "Women in Computer Science",
      description: "Stuff",
      tags: ["Technology", "Science", "Women"],
      members: [200000001, 200000002, 200000003, 200000004],
      events: [100001, 100002, 100003, 100004],
      organizers: [200000001]
    };

    const beekeepingData = {
      club_name: "Beekeeping Club",
      description: "Stuff",
      tags: ["Science", "Environment"],
      members: [200000001, 200000002, 200000003, 200000004],
      events: [100001, 100002, 100003, 100004],
      organizers: [200000001]
    };

    const userData = {
      managed_clubs: [wicsData, beekeepingData],
      first_name: "Anastasia",
      last_name: "Egorova",
      last_active: Date.now(),
      tags: ["Technology", "Women"]
    };

    return (
      <div className="event-management">
        <h1>Organizations</h1>
        <Button icon="plus" onClick={this.toggleDialog}>
          {" "}
          Add New Organization{" "}
        </Button>
        {userData.managed_clubs.map((club, index) => (
          <div className="image-gallery" key={club.club_name}>
            <div className="gallery-item">
              <img
                src={
                  index === 0
                    ? "https://static1.squarespace.com/static/58ed6670197aea8dbb07ec4e/t/5acac8faaa4a998f3f5dc351/1523239170320/"
                    : "https://pestworldforkids.org/media/12856/Small_bee-honeycomb.jpg?preset=fullWidth360"
                }
                alt={club.club_name}
                onClick={() => this.toggleDialog(club)}
              />
              <div className="description">{club.club_name}</div>
            </div>
          </div>
        ))}
        <Dialog
          autoFocus={true}
          isOpen={orgDialogOpened}
          onClose={this.handleCloseDialog}
          title="Organization Details"
          canEscapeKeyClose={true}
        >
          <div className={Classes.DIALOG_BODY}>
            <H5>Club Name</H5>
            <InputGroup
              large={true}
              round={true}
              fill={true}
              placeholder="Enter club name..."
              defaultValue={org.club_name}
            />
            <br />
            <H5>Description</H5>
            <textarea
              className="bp3-input bp3-round .modifier"
              dir="auto"
              rows="10"
              cols="1000"
              placeholder="Enter description..."
              defaultValue={org.description}
            />
            <br />
            <br />
            <H5>Tags</H5>
            <InputGroup
              large={true}
              round={true}
              fill={true}
              placeholder="Enter tags..."
              defaultValue={org.tags}
            />
            <br />
            <H5>Members</H5>
            <textarea
              className="bp3-input bp3-round .modifier"
              dir="auto"
              rows="10"
              cols="1000"
              placeholder="Enter members..."
              defaultValue={org.members}
            />
            <br />
            <br />
            <H5>Events</H5>
            <textarea
              className="bp3-input bp3-round .modifier"
              dir="auto"
              rows="10"
              cols="1000"
              placeholder="Enter events..."
              defaultValue={org.events}
            />
            <br />
            <br />
            <H5>Organizers</H5>
            <InputGroup
              large={true}
              round={true}
              fill={true}
              placeholder="Enter club organizers..."
              defaultValue={org.organizers}
            />
            <br />
          </div>
          <div className={Classes.DIALOG_FOOTER}>
            <div className={Classes.DIALOG_FOOTER_ACTIONS}>
              <Button onClick={this.handleCloseDialog}>Delete</Button>
              <AnchorButton
                intent={Intent.SUCCESS}
                href="https://www.palantir.com/palantir-foundry/"
                target="_blank"
              >
                Save
              </AnchorButton>
            </div>
          </div>
        </Dialog>
      </div>
    );
  }
}
