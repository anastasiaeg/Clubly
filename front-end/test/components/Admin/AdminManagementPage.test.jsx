import React from 'react';
import renderer from 'react-test-renderer';

import AdminManagementPage from '../../../src/components/Admin/AdminManagementPage';

describe('Admin Management Page', () => {
    test('should render', () => {
        const component = renderer.create(
            <AdminManagementPage/>
        )
        .toJSON();
        expect(component).toMatchSnapshot();
    });
})