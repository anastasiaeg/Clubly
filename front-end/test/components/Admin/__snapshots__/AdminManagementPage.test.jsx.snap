// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Management Page should render 1`] = `
<div
  className="event-management"
>
  <h1>
    Organizations
  </h1>
  <button
    className="bp3-button"
    disabled={undefined}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    tabIndex={undefined}
    type="button"
  >
    <span
      className="bp3-icon bp3-icon-plus"
      icon="plus"
    >
      <svg
        data-icon="plus"
        fill={undefined}
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          plus
        </desc>
        <path
          d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
          fillRule="evenodd"
        />
      </svg>
    </span>
    <span
      className="bp3-button-text"
    >
       Add New Organization 
    </span>
  </button>
  <div
    className="image-gallery"
  >
    <div
      className="gallery-item"
    >
      <img
        alt="Women in Computer Science"
        onClick={[Function]}
        src="https://picsum.photos/300/200/?random"
      />
      <div
        className="description"
      >
        Women in Computer Science
      </div>
    </div>
  </div>
  <div
    className="image-gallery"
  >
    <div
      className="gallery-item"
    >
      <img
        alt="Beekeeping Club"
        onClick={[Function]}
        src="https://picsum.photos/300/200/?random"
      />
      <div
        className="description"
      >
        Beekeeping Club
      </div>
    </div>
  </div>
</div>
`;
